src/models/piranha_arguments.rs

```diff
@@ -283,14 +292,31 @@ impl PiranhaArgumentsBuilder {
   /// * parse `piranha_arguments.toml` (if it exists)
   /// * merge the two PiranhaArguments
   pub fn build(&self) -> PiranhaArguments {
-    let _arg = &self.create().unwrap();
+    let mut _arg = self.create().unwrap();
+    let piranha_language = _arg.language();
+    let built_in_rules = RuleGraphBuilder::default()
+      .edges(piranha_language.edges().clone().unwrap_or_default().edges)
+      .rules(piranha_language.rules().clone().unwrap_or_default().rules)
+      .build();
+
+    let user_defined_rules = read_user_config_files(_arg.path_to_configurations());
 
-    let path_to_toml =
+    let rule_graph = built_in_rules.merge(&user_defined_rules);
+
+    let arg_rg = PiranhaArgumentsBuilder::default()
+      .rule_graph(rule_graph)
+      .create()
+      .unwrap();
+
+    _arg = _arg.merge(arg_rg);
```

Comment:
Not knowledgable enough about Rust to know if this is idiomatic, but this feels like an awkward way to just pass the `rule_graph` to the `PiranhaArguments` being built. Its creating a different instance of `PiranhaArguments` with partial configuration and then sort of combining it into this one, no?

Reply:
Addressed